### PR TITLE
[FIX] account_edi_proxy_client, l10n_it_edi_sdicoop: use test server, fix retry after notifica scarto

### PR DESCRIPTION
**Pull request by dbkosky**
The l10n_it_edi_sdicoop module makes use of an odoo iap server. There is a similar sever which points to the test channel for the Italian edi system. This PR adapts and existing config parameter (account_edi_proxy_client.demo) so that it can be one of three states:
'prod' or False (when the config parameter doesn't exist): use the production iap server, which points to the production channel. 
'test': use the test iap server, which points to the test channel
'demo' (or anthing else that's not the above two): don't send anything and simulate the results of the Italian edi process 

This PR also includes a fix that allows users to actually retry sending their invoices when the previous attempt fails (notificaScarto).

task-id: 2764978

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by dbkosky** [comment]
@jco-odoo

File: addons/l10n_it_edi_sdicoop/models/account_edi_format.py [hunk 1]
```diff
@@ -136,7 +137,7 @@ def _l10n_it_post_invoices_step_1(self, invoices):
             })
             invoice.l10n_it_edi_attachment_id = attachment
 
-            if invoice._is_commercial_partner_pa():
+            if len(invoice.commercial_partner_id.l10n_it_pa_index or '') == 6:
```
Comment (jco-odoo) [inline addons/l10n_it_edi_sdicoop/models/account_edi_format.py:140]: @dbkosky Why this change?

File: addons/l10n_it_edi_sdicoop/models/account_edi_format.py [hunk 2]
```diff
@@ -153,13 +154,15 @@ def _l10n_it_post_invoices_step_1(self, invoices):
                 'error': _("You must accept the terms and conditions in the settings to use FatturaPA."),
                 'blocking_level': 'error'} for invoice in invoices}
 
-        if self.env['ir.config_parameter'].get_param('account_edi_proxy_client.demo', False):
-            responses = {filename: {'id_transaction': 'demo'} for invoice in invoices}
-        else:
+        demo_state = self.env['ir.config_parameter'].get_param('account_edi_proxy_client.demo', False)
```
Comment (jco-odoo) [inline addons/l10n_it_edi_sdicoop/models/account_edi_format.py:157]: we could also have a tiny method _l10n_it_get_demo_state api.model which could return test/demo/prod so we don't need to repeat if demo_state in...  (I first thought it as not too bad, but we have it like 7 times)

File: addons/l10n_it_edi_sdicoop/models/account_edi_format.py [hunk 3]
```diff
@@ -223,13 +227,11 @@ def _l10n_it_post_invoices_step_2(self, invoices):
             xml = proxy_user._decrypt_data(response['file'], response['key'])
             response_tree = etree.fromstring(xml)
             if state == 'ricevutaConsegna':
-                if invoice._is_commercial_partner_pa():
-                    to_return[invoice] = {'error': _('The invoice has been succesfully transmitted. The addressee has 15 days to accept or reject it.')}
-                else:
-                    to_return[invoice] = {'attachment': invoice.l10n_it_edi_attachment_id, 'success': True}
+                to_return[invoice] = {'error': _('The invoice has been succesfully transmitted. The addressee has 15 days to accept or reject it.')}
```
Comment (jco-odoo) [inline addons/l10n_it_edi_sdicoop/models/account_edi_format.py:230]: Why?  The logic above was correct.  If the partner is from the government, he can still refuse within 15 days, otherwise not.

File: addons/l10n_it_edi_sdicoop/models/account_edi_format.py [hunk 4]
```diff
@@ -223,13 +227,11 @@ def _l10n_it_post_invoices_step_2(self, invoices):
...
             elif state == 'notificaScarto':
                 errors = [element.find('Descrizione').text for element in response_tree.xpath('//Errore')]
-                to_return[invoice] = {'error': self._format_error_message(_('The invoice has been refused by the Exchange System'), errors), 'blocking_level': 'error'}
+                to_return[invoice] = {'error': self._format_error_message(_('The invoice has been refused by the Exchange System'), errors), 'blocking_level': 'error'}# 'l10n_it_edi_transaction': False
```
Comment (jco-odoo) [inline addons/l10n_it_edi_sdicoop/models/account_edi_format.py:233]: comment not needed

**Review by jco-odoo — COMMENTED** [review]
@dbkosky Could you check?

File: addons/l10n_it_edi_sdicoop/models/account_edi_format.py (hunk 1 above)
Comment (dbkosky) [reply addons/l10n_it_edi_sdicoop/models/account_edi_format.py:140]: Sorry I think some of these were a result of rebasing on the wrong branch

**Comment by jco-odoo** [comment]
@robodoo r+ rebase-ff
